### ISSUE TITLE
Find in Files doesn't load

### ISSUE THREAD

**Issue by ynraza** [rstudio/rstudio#12657]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.12.0+353 (2022.12.0+353)
    OS Version      : macOS Catalina 10.15.6
    R Version       : 4.2.2 (2022-10-31)

### Steps to reproduce the problem

Use 'find in files' function to search for a word.

### Describe the problem in detail

Find in files function doesn't work. It continues to "search" without finding anything or displaying results, even if searching in a path that contains specific files that would have the matching word.

<img width="790" alt="Screenshot 2023-02-01 at 15 47 17" src="https://user-images.githubusercontent.com/75734732/216092505-ab4666e8-6571-4c9d-944c-95c7f32ffce9.png">

Enabling RSTUDIO_GREP_DEBUG shows the following:
> Sys.setenv(RSTUDIO_GREP_DEBUG = 1)
grep '--binary-files=without-match' '-rHn' '--color=always' '--devices=skip' '-i' '-F' '-f' '/var/folders/0v/yk1jj65d6msbbx6m_kzn25c80000gn/T/Rtmpuo1FMS/rs_grep8f253f88b7dd.txt' '--include=*.r' '--include=*.R'

### Describe the behavior you expected

Find in files was working in previous versions of R Studio. Would display results/files.  

- [X] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [X] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
Can you please share a screenshot of what your search query looks like? For example:

<img width="416" alt="Screenshot 2023-02-01 at 9 40 34 AM" src="https://user-images.githubusercontent.com/1976582/216120631-720d99bd-c8f4-4b81-a7c4-792f8b507fda.png">

The include statement here looks incorrect:

```
'--include=.r' '--include=.R'
```

as that should be `--include=*.r --include=*.R`, but I wasn't able to reproduce that.

**Comment by ynraza** [comment]
Sorry just to follow up on this I have also tried using custom filter to include *.r  *.R and grep debug shows the following:

>Sys.setenv(RSTUDIO_GREP_DEBUG = 1)
grep '--binary-files=without-match' '-rHn' '--color=always' '--devices=skip' '-i' '-F' '-f' '/var/folders/0v/yk1jj65d6msbbx6m_kzn25c80000gn/T/RtmpSoJv37/rs_grep968b8c18f98.txt' '--include=*.r' '--include=*.R'

<img width="405" alt="Screenshot 2023-02-02 at 11 31 53" src="https://user-images.githubusercontent.com/75734732/216314030-99f27550-e0f2-41db-b8c2-1cdd14bad1f7.png">

**Comment by ynraza** [comment]
Sorry just to follow up on this the include statement includes *.r and *.R, not sure why it didnt show above.

<img width="778" alt="Screenshot 2023-02-02 at 11 49 51" src="https://user-images.githubusercontent.com/75734732/216317371-a15bc2f5-2a92-42b7-9714-d3264d7768c1.png">


<img width="405" alt="Screenshot 2023-02-02 at 11 31 53" src="https://user-images.githubusercontent.com/75734732/216317400-f443ded1-0e76-4f0e-a433-e47081de5ed6.png">

Results still don't load.

**Comment by kevinushey** [comment]
Oops, I see now that your search query did include `*.r` and `*.R`; it was just GitHub's markdown rendering that turned those into italics. 😅 

Then ... my only guess is that something in that folder is getting hung up while trying to `grep`? Do you have a large number of files / folders in that location? Do you see the same issue if you grep within a smaller folder?

**Comment by ynraza** [comment]
Thanks. I've tired with a smaller folder and still get the same issue unfortunately.

**Comment by kevinushey** [comment]
Can you also confirm what the output of `Sys.which("grep")` is, in case an unexpected copy of `grep` is being used?

Do you have any better luck if you try to run `grep` "by hand" in a Terminal instance; for example:

```
grep -rHn --include="*.R" --include="*.r" ggplot
```

What do you see if you run that in one of the folders you're testing?

**Comment by ynraza** [comment]
Hi,

Thanks. Output of Sys.which("grep")

> Sys.which("grep")
           grep 
"/usr/bin/grep" 

Testing grep in the Terminal seems to work

>$ grep -rHn --include="*.R" --include="*.r" ggplot .
./test_script.R:19:library(ggplot2)
./test_script.R:24:ggplot(df, aes(x=year, y=var1))+
./test_script.R:33:ggplot(df, aes(x=year, y=var2))+
./test_script.R:42:ggplot(df, aes(x=year, y=var3))+

Have tired this in larger folders also and grep will print the output.

**Comment by kevinushey** [comment]
Interesting; I'm pretty confused then. Can you provide a diagnostics report, as described in https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report?

**Comment by skyeturriff** [comment]
I believe [this support customer](https://rstudioide.zendesk.com/agent/tickets/84116) may also be experiencing this issue. 

Running with grep debug reports no errors. Results are in fact found and output to the R console, but they are not always displayed in the Find in Files pane (see screenshots from customer).

The customer also notes:
>  The Find in Files does work when the strace is enabled, but stopped after it was cancelled.

> Output 1: nothing in Find in Files and stop sign button is on
Output 2: after clicked on red stop sign then refresh. Find in Files shows correct results

In discussing with Jeff it was thought this could potentially be a timing issue, where the callbacks to clear the old results and render the new results are occasionally getting out of order.

**Comment by kevinushey** [comment]
Thanks; I took a look at the request log. Here's the "success" case:

`````
{
  "method": "get_events",
  "params": [
    96
  ],
  "clientId": "3c6016b8-0512-4c0a-9e36-6bebd042c66b",
  "clientVersion": "02ace23e263335c78e1df5b0892e48913e5fcecf"
}
{
  "result": [
    {
      "id": 97,
      "type": "find_result",
      "data": {
        "handle": "d5ee292eb9464a958aec2b58e14fcaca",
        "results": {
          "file": [
            "~/vr.test/R/gt_image.R",
            "~/vr.test/R/gt_image.R",
            "~/vr.test/R/gt_image.R"
          ],
          "line": [
            20,
            23,
            25
          ],
          "lineValue": [
            "economics_sample_df <- slice_sample(economics, n = 10, replace = F) %>% arrange(date)",
            "economics_sample_gt <- economics_sample_df %>% gt() |>",
            "gtsave(filename = \"R/economics_data_sample.png\", expand = 10)"
          ],
          "matchOn": [
            [
              0
            ],
            [
              0,
              23
            ],
            [
              21
            ]
          ],
          "matchOff": [
            [
              10
            ],
            [
              10,
              33
            ],
            [
              31
            ]
          ],
          "replaceMatchOn": [
            [],
            [],
            []
          ],
          "replaceMatchOff": [
            [],
            [],
            []
          ],
          "errors": [
            [],
            [],
            []
          ]
        }
      }
    },
    {
      "id": 98,
      "type": "find_operation_ended",
      "data": "d5ee292eb9464a958aec2b58e14fcaca"
    },
    {
      "id": 99,
      "type": "session_suspend_blocked",
      "data": []
    }
  ],
  "ep": "false"
}

{
  "method": "begin_find",
  "params": [
    "economics_",
    false,
    false,
    true,
    "~/vr.test",
    [
      "*.r",
      "*.R"
    ],
    true,
    false,
    []
  ],
  "clientId": "3c6016b8-0512-4c0a-9e36-6bebd042c66b",
  "clientVersion": "02ace23e263335c78e1df5b0892e48913e5fcecf"
}
{
  "result": "d5ee292eb9464a958aec2b58e14fcaca",
  "ep": "false"
}
`````

And here's the "failed" case:

`````
{
  "method": "get_events",
  "params": [
    115
  ],
  "clientId": "3c6016b8-0512-4c0a-9e36-6bebd042c66b",
  "clientVersion": "02ace23e263335c78e1df5b0892e48913e5fcecf"
}
{
  "result": [
    {
      "id": 116,
      "type": "find_result",
      "data": {
        "handle": "e12da7c087484f6d83c218ac384c9d7e",
        "results": {
          "file": [
            "~/vr.test/R/gt_image.R",
            "~/vr.test/R/gt_image.R",
            "~/vr.test/R/gt_image.R"
          ],
          "line": [
            20,
            23,
            25
          ],
          "lineValue": [
            "economics_sample_df <- slice_sample(economics, n = 10, replace = F) %>% arrange(date)",
            "economics_sample_gt <- economics_sample_df %>% gt() |>",
            "gtsave(filename = \"R/economics_data_sample.png\", expand = 10)"
          ],
          "matchOn": [
            [
              0
            ],
            [
              0,
              23
            ],
            [
              21
            ]
          ],
          "matchOff": [
            [
              10
            ],
            [
              10,
              33
            ],
            [
              31
            ]
          ],
          "replaceMatchOn": [
            [],
            [],
            []
          ],
          "replaceMatchOff": [
            [],
            [],
            []
          ],
          "errors": [
            [],
            [],
            []
          ]
        }
      }
    },
    {
      "id": 117,
      "type": "find_operation_ended",
      "data": "e12da7c087484f6d83c218ac384c9d7e"
    },
    {
      "id": 118,
      "type": "session_suspend_blocked",
      "data": []
    }
  ],
  "ep": "false"
}

{
  "method": "begin_find",
  "params": [
    "economics_",
    false,
    false,
    true,
    "~/vr.test",
    [
      "*.r",
      "*.R"
    ],
    true,
    false,
    []
  ],
  "clientId": "3c6016b8-0512-4c0a-9e36-6bebd042c66b",
  "clientVersion": "02ace23e263335c78e1df5b0892e48913e5fcecf"
}
{
  "result": "e12da7c087484f6d83c218ac384c9d7e",
  "ep": "false"
}
`````

The confusing bit is that the events logged here are essentially the same -- both cases seem to report find results, and both have events logged in the same order. It is perhaps strange that the find results are being logged before the 'begin_find' RPC, but I'm not sure what to make of that.

So there must be something else going on that we're missing?

**Comment by jonvanausdeln** [comment]
This appears to be fixed in latest Daily build [2023.12.0-daily+262](https://dailies.rstudio.com/version/2023.12.0-daily+262)

@ynraza, if you'd like to try it out you can get the latest daily builds [here](https://dailies.rstudio.com/). Please note that daily builds are intended for testing only and not production environments.
